Row.c

```diff
@@ -314,79 +314,62 @@ void Row_printTime(RichString* str, unsigned long long totalHundredths, bool col
-         RichString_appendnAscii(str, dayColor, buffer, len);
-         len = xSnprintf(buffer, sizeof(buffer), "%1dh ", hoursLeft);
-         RichString_appendnAscii(str, hourColor, buffer, len);
-      }
-   } else if (hours >= 100) {
-      int minutesLeft = totalSeconds / 60 - hours * 60;
-
-      if (minutesLeft >= 10) {
-         len = xSnprintf(buffer, sizeof(buffer), "%4lluh", hours);
-         RichString_appendnAscii(str, hourColor, buffer, len);
-         len = xSnprintf(buffer, sizeof(buffer), "%2dm ", minutesLeft);
-         RichString_appendnAscii(str, baseColor, buffer, len);
-      } else {
-         len = xSnprintf(buffer, sizeof(buffer), "%5lluh", hours);
-         RichString_appendnAscii(str, hourColor, buffer, len);
-         len = xSnprintf(buffer, sizeof(buffer), "%1dm ", minutesLeft);
-         RichString_appendnAscii(str, baseColor, buffer, len);
-      }
-   } else if (hours > 0) {
-      len = xSnprintf(buffer, sizeof(buffer), "%2lluh", hours);
+   int baseColor = CRT_colors[PROCESS];
+
+   unsigned long long totalSeconds = totalHundredths / 100;
+   unsigned long long totalMinutes = totalSeconds / 60;
+   unsigned long long totalHours = totalMinutes / 60;
+   unsigned int seconds = totalSeconds - totalMinutes * 60;
+   unsigned int minutes = totalMinutes - totalHours * 60;
+
+   if (totalMinutes < 60) {
+      unsigned int hundredths = totalHundredths - totalSeconds * 100;
+      len = xSnprintf(buffer, sizeof(buffer), "%2llu:%02u.%02u ", totalMinutes, seconds, hundredths);
+      RichString_appendnAscii(str, baseColor, buffer, len);
+      return;
+   }
+   if (totalHours < 24) {
+      len = xSnprintf(buffer, sizeof(buffer), "%2lluh", totalHours);
       RichString_appendnAscii(str, hourColor, buffer, len);
-      len = xSnprintf(buffer, sizeof(buffer), "%02d:%02d ", minutes, seconds);
+      len = xSnprintf(buffer, sizeof(buffer), "%02u:%02u ", minutes, seconds);
       RichString_appendnAscii(str, baseColor, buffer, len);
-   } else {
-      len = xSnprintf(buffer, sizeof(buffer), "%2d:%02d.%02d ", minutes, seconds, hundredths);
+      return;
+   }
+
+   unsigned long long totalDays = totalHours / 24;
+   unsigned int hours = totalHours - totalDays * 24;
+   if (totalDays < 10) {
+      len = xSnprintf(buffer, sizeof(buffer), "%1llud", totalDays);
+      RichString_appendnAscii(str, dayColor, buffer, len);
+      len = xSnprintf(buffer, sizeof(buffer), "%2uh", hours);
+      RichString_appendnAscii(str, hourColor, buffer, len);
+      len = xSnprintf(buffer, sizeof(buffer), "%2um ", minutes);
       RichString_appendnAscii(str, baseColor, buffer, len);
+      return;
+   }
+   if (totalDays < /* Ignore leap years */365) {
+      len = xSnprintf(buffer, sizeof(buffer), "%4llud", totalDays);
+      RichString_appendnAscii(str, dayColor, buffer, len);
+      len = xSnprintf(buffer, sizeof(buffer), "%2uh ", hours);
+      RichString_appendnAscii(str, hourColor, buffer, len);
```

Comment:
Avoid spaces in the middle …
```suggestion
      len = xSnprintf(buffer, sizeof(buffer), "%02uh ", hours);
      RichString_appendnAscii(str, hourColor, buffer, len);
```